x-pack/plugins/uptime/public/components/monitor/synthetics/step_detail/waterfall/waterfall_chart_container.tsx

```diff
@@ -59,12 +60,35 @@ export const WaterfallChartContainer: React.FC<Props> = ({ checkGroup, stepIndex
           </EuiFlexItem>
         </EuiFlexGroup>
       )}
-      {networkEvents && !networkEvents.loading && networkEvents.events.length > 0 && (
-        <WaterfallChartWrapper
-          data={extractItems(networkEvents.events)}
-          total={networkEvents.total}
-        />
-      )}
+      {networkEvents &&
+        !networkEvents.loading &&
+        networkEvents.events.length > 0 &&
+        networkEvents.isWaterfallSupported && (
+          <WaterfallChartWrapper
+            data={extractItems(networkEvents.events)}
+            total={networkEvents.total}
+          />
+        )}
+      {networkEvents &&
+        !networkEvents.loading &&
+        networkEvents.events.length > 0 &&
+        !networkEvents.isWaterfallSupported && (
```

Comment:
can we please clean these conditions a bit? maybe extract them to a variable at top and use those variables. something like
```

const showWaterFall = networkEvents && !networkEvents.loading &&
        networkEvents.events.length > 0 &&
        networkEvents.isWaterfallSupported;
```